integration_test/itest/image.go

```diff
@@ -1,13 +1,54 @@
 package itest
 
-import "context"
+import (
+	"context"
+	"strings"
+
+	"github.com/telepresenceio/telepresence/v2/pkg/dos"
+)
 
 type Image struct {
 	Name     string `json:"name"`
 	Tag      string `json:"tag,omitempty"`
 	Registry string `json:"registry,omitempty"`
 }
 
+func (img *Image) FQName() string {
+	nb := strings.Builder{}
+	if img.Registry != "" {
+		nb.WriteString(img.Registry)
+		nb.WriteByte('/')
+	}
+	nb.WriteString(img.Name)
+	if img.Tag != "" {
+		nb.WriteByte(':')
+		nb.WriteString(img.Tag)
+	}
+	return nb.String()
+}
+
+func ImageFromEnv(ctx context.Context, env, defaultTag, defaultRegistry string) *Image {
+	if imgQN, ok := dos.LookupEnv(ctx, env); ok {
+		img := new(Image)
+		i := strings.LastIndexByte(imgQN, '/')
+		if i >= 0 {
+			img.Registry = imgQN[:i]
+			imgQN = imgQN[i+1:]
+		} else {
+			img.Registry = defaultRegistry
+		}
+		if i = strings.IndexByte(imgQN, ':'); i > 0 {
+			img.Name = imgQN[:i]
+			img.Tag = imgQN[i+1:]
+		} else {
+			img.Name = imgQN
+			img.Tag = defaultTag
+		}
+		return img
+	}
+	return nil
+}
```

Comment:
are there docker client libraries we can maybe use to parse this? i just get queasy at this kind of character-wise manipulation of a formatted string